packages/core/src/application.ts

```diff
@@ -98,11 +103,23 @@ export class Application extends Context implements LifeCycleObserver {
     // Make options available to other modules as well.
     this.bind(CoreBindings.APPLICATION_CONFIG).to(this.options);
 
-    const shutdownConfig = this.options.shutdown ?? {};
-    this.setupShutdown(
-      shutdownConfig.signals ?? ['SIGTERM'],
-      shutdownConfig.gracePeriod,
-    );
+    this._shutdownOptions = {signals: ['SIGTERM'], ...this.options.shutdown};
+  }
+
+  /**
+   * Wrap the debug statement so that it always print out the context name
+   * as the prefix
+   * @param args - Arguments for the debug
+   */
+  private debugWithName(...args: unknown[]) {
```

Comment:
Should the Context class prefix debug logs with the context name too?

Reply:
I made the context.debugWithName() protected.